dockstore-webservice/src/main/java/io/dockstore/webservice/languages/NextFlowHandler.java

```diff
@@ -111,31 +111,143 @@ private ConfigObject getConfigObject(String content) {
     }
 
     /**
-     * Grabs the path in the AST to the desired child node
-     *
-     * @param ast
-     * @param keyword
-     * @param groovyTokenType
-     * @return
+     * Returns the first AST found with some keyword as text
+     * @param ast An AST
+     * @param keyword Text to search node for (exact match)
+     * @param compareChild If true will check first child for keyword, if false will check current node
+     * @return AST with some keyword as text
      */
-    private List<GroovySourceAST> extractTargetFromAST(GroovySourceAST ast, String keyword, int groovyTokenType) {
+    private GroovySourceAST getFirstAstWithKeyword(GroovySourceAST ast, String keyword, boolean compareChild) {
         if (ast == null) {
             return null;
         }
-        if (ast.getType() == groovyTokenType && ast.getText().equals(keyword)) {
-            return Lists.newArrayList(ast);
+
+        if (compareChild) {
+            if (ast.getFirstChild() != null && Objects.equals(ast.getFirstChild().getText(), keyword)) {
+                return ast;
+            }
+        } else {
+            if (Objects.equals(ast.getText(), keyword)) {
+                return ast;
+            }
         }
 
         int numChildren = ast.getNumberOfChildren();
+        GroovySourceAST subtree = null;
+
         for (int i = 0; i < numChildren; i++) {
             GroovySourceAST groovySourceAST = ast.childAt(i);
-            List<GroovySourceAST> target = extractTargetFromAST(groovySourceAST, keyword, groovyTokenType);
-            if (target != null) {
-                target.add(ast);
-                return target;
+            subtree = getFirstAstWithKeyword(groovySourceAST, keyword, compareChild);
+            if (subtree != null) {
+                break;
             }
         }
-        return null;
+
+        if (subtree == null && ast.getNextSibling() != null) {
+            subtree = getFirstAstWithKeyword((GroovySourceAST)ast.getNextSibling(), keyword, compareChild);
+        }
+        return subtree;
+    }
+
+
+    /**
+     * Given an AST for an EXPR will return the text name
+     * @param exprAST AST of an EXPR
+     * @return Input channel name
+     */
+    private String getInputChannelNameForEXPR(GroovySourceAST exprAST) {
+        return exprAST == null ? null : exprAST.getFirstChild().getFirstChild().getNextSibling().getFirstChild().getText();
```

Comment:
Are we catching exceptions properly? 
This code is pretty fragile (my fault in the beginning)

Reply:
We catch all of these IOException | TokenStreamException | RecognitionException e.